dwave/optimization/generators.py

```diff
@@ -407,6 +409,243 @@ class as the decision variable being optimized, with permutations of its
+    be served after time_window_open and before time_window_close. Vehicles have a
+    specified delivery capacity, and on the routes to locations and then back
+    to the depot, no vehicle is allowed to exceed its carrying capacity.
+
+    Args:
+        demand:
+            Customer demand, as an |array-like|_. The first element is the depot and must
+            be zero.
+        number_of_vehicles:
+            Number of available vehicles, as an integer.
+        vehicle_capacity:
+            Maximum capacity for any vehicle. The total delivered demand by any
+            vehicle on any route must not exceed this value.
+        time_distances:
+            time_distances between **all** the problem's locations, as an |array-like|_
+            of positive numbers, including both customer sites and the depot.The first
+            row and colum of the distance matrix are customer distances form the depot.
+        time_window_open:
+            The opening time of each customer, as an |array-like|_. The first element is
+            the depot.
+        time_window_close:
+            The closing time of each customer, as an |array-like|_. The first element is
+            the depot.
+        service_time:
+            The time it takes to serve each customer, as an |array-like|_. The first element
+            is the depot and must be zero.
+
+    Returns:
+        A model encoding the CVRPTW problem.
+
+    Notes:
+        The model uses a :class:`~dwave.optimization.model.Model.disjoint_lists`
+        class as the decision variable being optimized, with permutations of its
+        sublist representing various itineraries for each vehicle.
+    """
+
+    time_distances = np.asarray(time_distances)
+    if not isinstance(number_of_vehicles, int):
+        raise ValueError("`number_of_vehicles` must be an integer.")
+
+    if number_of_vehicles < 1:
+        raise ValueError("`number_of_vehicles` must be at least 1."
+                         f" Got {number_of_vehicles}.")
+
+    if vehicle_capacity <= 0:
+        raise ValueError("`vehicle_capacity` must be a positive number."
+                         f" Got {vehicle_capacity}.")
+
+    demand = _require("demand", demand, dtype=float, ndim=1, nonnegative=True)
+
+    if demand.sum() > number_of_vehicles * vehicle_capacity:
+        raise ValueError(f"Total demand, {demand.sum()}, exceeds the "
+                         f"total capacity, {number_of_vehicles * vehicle_capacity}.")
+
+    time_distances_array = _require("time_distances", time_distances, ndim=2, nonnegative=True, square=True)
+
+    time_window_open = _require("time_window_open", time_window_open, dtype=float, ndim=1, nonnegative=True)
+    time_window_close = _require("time_window_close", time_window_close, dtype=float, ndim=1, nonnegative=True)
+    service_time = _require("service_time", service_time, dtype=float, ndim=1, nonnegative=True)
+
+    if len(time_distances_array) < 2:
```

Comment:
`len()` in this case returns the size of the first dimension. IMO we should use either `.size` which returns the number of elements or `.shape[0]`. The error message should probably reflect that as well.